.github/workflows/linuxtests.yml

```diff
@@ -0,0 +1,58 @@
+name: Linux Tests
+
+on:
+  push:
+    branches:
+      - master
+      - maintenance/**
+  pull_request:
+    branches:
+      - master
+      - maintenance/**
+
+jobs:
+  Python_3.7_Debug
+    name: Python 3.7 Debug
+    runs-on: ubuntu-18.04
+    env:
+      USE_DEBUG=python3.7-dbg
+      TESTMODE=fast
+      NUMPYSPEC="--upgrade numpy"
+    steps:
+    - uses: actions/checkout@v2
+    - name: Create a venv for python-dbg runs-on
+      run: |
+        virtualenv --python=$USE_DEBUG venv
+        source venv/bin/activate
+        apt install python3.7-dbg python3.7-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache swig libmpc-dev
+    - name: Configure env vars and python packages
+      run: |
+        free -m
+        export PATH=/usr/lib/ccache:$PATH
+        ccache --zero-stats
+        export CCACHE_COMPRESS=1
+        python --version # just to check
+        export NPY_NUM_BUILD_JOBS=2
+        uname -a
+        df -h
+        ulimit -a
+        set -e -o pipefail
+        mkdir builds
+        cd builds
+        python -c 'import sys; print("Python debug build:", hasattr(sys, "gettotalrefcount"))'
+        pip install --upgrade pip setuptools wheel $OTHERSPEC cython pytest pytest-xdist mpmath argparse gmpy2 pybind11
+        pip install --upgrade $NUMPYSPEC
+        pip install
+        pip uninstall -y nose
+        ccache -s
+        cd ..
+    - name: Building SciPy
+      run: python -u runtests.py -g -j2 --build-only $COVERAGE $USE_WHEEL_BUILD
```

Comment:
these environment variables don't exist. I would remove the environment variables anyway.

Reply:
see above, I'll clean up the jobs anyways since these were heavily adapted to Travis workflow